llvm/test/SYCLLowerIR/ConditionalCallOnDevice/conditional-call-on-device-1.ll

```diff
@@ -0,0 +1,37 @@
+; RUN: opt -passes=sycl-conditional-call-on-device < %s -S | FileCheck %s
+
+%class.anon = type { ptr addrspace(4) }
+%"struct.std::integer_sequence.3" = type { i8 }
+
+define internal spir_func void @_ZN4sycl3_V13ext6oneapi12experimental6detail38call_if_on_device_conditionally_helperIZZZ4mainENKUlRNS0_7handlerEE_clES7_ENKUlvE_clEvEUlvE_JLin2ELi251660032EEEEvT_St16integer_sequenceIiJXspT0_EEE(ptr noundef byval(%class.anon) align 8 %fn, ptr noundef byval(%"struct.std::integer_sequence.3") align 1 %0) #2 !srcloc !0 {
```

Comment:
This is clearly produced from a real life example, but at the same time a lot of LLVM IR was thrown away to make the test smaller. Considering the latter, we should rename those functions as well to improve readability of the test.

We should probably keep `_ZN4sycl3_V13ext6oneapi12experimental6detail38call_if_on_device_conditionally_helper` prefix, but the rest can be dropped